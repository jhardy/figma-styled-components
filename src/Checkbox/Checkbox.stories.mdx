import { Checkbox } from "./Checkbox";
import { Story, Meta } from "@storybook/addon-docs/blocks";

<Meta title="MDX|Checkbox" component={Checkbox} />

# Checkbox

<Story name="Checkbox">
  <div className="component-box">
    <Checkbox label="I am a checkbox click me" />
  </div>
</Story>

<div className="props-table">

## Props

| Prop Name | Type     | Default | Required | Description                              |
| :-------- | :------- | :-----: | :------: | :----------------------------------------|
| label     | string   |    -    |   yes    | The checkbox's label                     |
| value     | string   |    -    |   yes    | The checkbox's value                    |
| checked   | boolean  |  false  |    -     | If the checkbox is checked or not        |
| onChange  | function |    -    |    -     | Callback for checkbox's `onChange` event, returns an event object |

</div>
