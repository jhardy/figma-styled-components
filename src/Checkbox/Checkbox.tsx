import * as React from 'react'
import styled from 'styled-components'
import { Text } from '../Text'


const CheckboxMark = styled.span`
  position: absolute;
  top: 10px;
  left: 10px;
  width: 12px;
  height: 12px;
  border: 1px solid #000000;
  border-radius: 2px;
  background-color: #ffffff;
  box-sizing: border-box;

  &:after {
    position: absolute;
    width: 10px;
    height: 10px;
    content: "";
    background-image: url("data:image/svg+xml;utf8,%3Csvg%20fill%3D%22none%22%20height%3D%227%22%20viewBox%3D%220%200%208%207%22%20width%3D%228%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Cpath%20clip-rule%3D%22evenodd%22%20d%3D%22m1.17647%201.88236%201.88235%201.88236%203.76471-3.76472%201.17647%201.17648-4.94118%204.9412-3.05882-3.05884z%22%20fill%3D%22%23fff%22%20fill-rule%3D%22evenodd%22%2F%3E%3C%2Fsvg%3E");
    background-repeat: no-repeat;
    background-position: center center;
  }
`

export interface CheckboxContainerProps {
  checked?: boolean,
  value: string
  onChange?: ((event: React.ChangeEvent<HTMLInputElement>) => void)
}
const CheckboxContainerFacotry: React.FC<CheckboxContainerProps> = ({ onChange, checked, value, ...props }) => {
  const handleChange = (event: any) => {
    if (onChange) { onChange(event) }
  }
  return (
    <div {...props}>
      <input type='checkbox' onChange={handleChange} checked={checked} value={value} />
      <CheckboxMark />
    </div>
  )
}

const CheckboxContainer = styled(CheckboxContainerFacotry)`
  position: relative;
  width: 32px;
  height: 32px;

  input {
    position: absolute;
    width: 0;
    height: 0;
    opacity: 0;
  }

  input:checked + ${CheckboxMark} {
    border: 1px solid #18a0fb;
    background-color: #18a0fb;
  }

  input:checked ${CheckboxMark}:after {
    display: block;
  }
`


export interface CheckboxProps {
  onChange?: ((event: React.ChangeEvent<HTMLInputElement>) => void)
  checked?: boolean
  label: string
  value: string
}
const CheckboxFactory: React.FC<CheckboxProps> = ({ onChange, label, checked, value, ...props }) => {
  return (
    <label {...props}>
      <CheckboxContainer onChange={onChange} checked={checked} value={value} />
      <Text>{label}</Text>
    </label>
  )
}

export const Checkbox = styled(CheckboxFactory)`
  display: flex;
  flex-direction: row;
  height: 28px;
  cursor: default;
  align-items: center;
`
