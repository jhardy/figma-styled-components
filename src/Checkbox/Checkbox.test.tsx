import 'jest-styled-components'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

import { Checkbox } from './Checkbox'


test('A default Checkbox', () => {
  const tree = renderer.create(<Checkbox label='A checkbox' value='default' />).toJSON()
  expect(tree).toMatchSnapshot()
})


test('A checked Checkbox', () => {
  const tree = renderer.create(<Checkbox label='A checked checkbox' checked={true} value='checked' />).toJSON()
  expect(tree).toMatchSnapshot()
})


