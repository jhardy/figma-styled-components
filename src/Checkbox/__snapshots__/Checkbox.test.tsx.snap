// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A checked Checkbox 1`] = `
.c4 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: "Inter",sans-serif;
  color: rgba(0,0,0,0.8);
  font-weight: 400;
  font-size: 11px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 16px;
}

.c3 {
  position: absolute;
  top: 10px;
  left: 10px;
  width: 12px;
  height: 12px;
  border: 1px solid #000000;
  border-radius: 2px;
  background-color: #ffffff;
  box-sizing: border-box;
}

.c3:after {
  position: absolute;
  width: 10px;
  height: 10px;
  content: "";
  background-image: url("data:image/svg+xml;utf8,%3Csvg%20fill%3D%22none%22%20height%3D%227%22%20viewBox%3D%220%200%208%207%22%20width%3D%228%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Cpath%20clip-rule%3D%22evenodd%22%20d%3D%22m1.17647%201.88236%201.88235%201.88236%203.76471-3.76472%201.17647%201.17648-4.94118%204.9412-3.05882-3.05884z%22%20fill%3D%22%23fff%22%20fill-rule%3D%22evenodd%22%2F%3E%3C%2Fsvg%3E");
  background-repeat: no-repeat;
  background-position: center center;
}

.c1 {
  position: relative;
  width: 32px;
  height: 32px;
}

.c1 input {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

.c1 input:checked + .c2 {
  border: 1px solid #18a0fb;
  background-color: #18a0fb;
}

.c1 input:checked .c2:after {
  display: block;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 28px;
  cursor: default;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<label
  className="c0"
>
  <div
    className="c1"
  >
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
      value="checked"
    />
    <span
      className="c2 c3"
    />
  </div>
  <span
    className="c4"
  >
    A checked checkbox
  </span>
</label>
`;

exports[`A default Checkbox 1`] = `
.c4 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: "Inter",sans-serif;
  color: rgba(0,0,0,0.8);
  font-weight: 400;
  font-size: 11px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 16px;
}

.c3 {
  position: absolute;
  top: 10px;
  left: 10px;
  width: 12px;
  height: 12px;
  border: 1px solid #000000;
  border-radius: 2px;
  background-color: #ffffff;
  box-sizing: border-box;
}

.c3:after {
  position: absolute;
  width: 10px;
  height: 10px;
  content: "";
  background-image: url("data:image/svg+xml;utf8,%3Csvg%20fill%3D%22none%22%20height%3D%227%22%20viewBox%3D%220%200%208%207%22%20width%3D%228%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Cpath%20clip-rule%3D%22evenodd%22%20d%3D%22m1.17647%201.88236%201.88235%201.88236%203.76471-3.76472%201.17647%201.17648-4.94118%204.9412-3.05882-3.05884z%22%20fill%3D%22%23fff%22%20fill-rule%3D%22evenodd%22%2F%3E%3C%2Fsvg%3E");
  background-repeat: no-repeat;
  background-position: center center;
}

.c1 {
  position: relative;
  width: 32px;
  height: 32px;
}

.c1 input {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

.c1 input:checked + .c2 {
  border: 1px solid #18a0fb;
  background-color: #18a0fb;
}

.c1 input:checked .c2:after {
  display: block;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 28px;
  cursor: default;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<label
  className="c0"
>
  <div
    className="c1"
  >
    <input
      onChange={[Function]}
      type="checkbox"
      value="default"
    />
    <span
      className="c2 c3"
    />
  </div>
  <span
    className="c4"
  >
    A checkbox
  </span>
</label>
`;
